tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/connect/ValidateAMQPErrorsTest.java

```diff
@@ -0,0 +1,643 @@
+
+      AssertionLoggerHandler.startCapture(true);
+
+      ProtonServerOptions serverOptions = new ProtonServerOptions();
+
+      AtomicInteger countOpen = new AtomicInteger(0);
+      CyclicBarrier startFlag = new CyclicBarrier(2);
+      CountDownLatch blockBeforeOpen = new CountDownLatch(1);
+      AtomicInteger disconnects = new AtomicInteger(0);
+      AtomicInteger messagesReceived = new AtomicInteger(0);
+      AtomicInteger errors = new AtomicInteger(0);
+
+      ConcurrentHashSet<ProtonConnection> connections = new ConcurrentHashSet<>();
+
+      mockServer = new MockServer(vertx, serverOptions, null, serverConnection -> {
+         serverConnection.disconnectHandler(c -> {
+            disconnects.incrementAndGet(); // number of retries
+            connections.remove(c);
+         });
+         serverConnection.openHandler(serverSender -> {
+            serverConnection.closeHandler(x -> {
+               serverConnection.close();
+               connections.remove(serverConnection);
+            });
+            serverConnection.open();
+            connections.add(serverConnection);
+         });
+         serverConnection.sessionOpenHandler((s) -> {
+            s.open();
+         });
+         serverConnection.senderOpenHandler((x) -> {
+            x.open();
+         });
+         serverConnection.receiverOpenHandler((x) -> {
+            if (countOpen.incrementAndGet() > 2) {
+               if (countOpen.get() == 3) {
+                  try {
+                     startFlag.await(10, TimeUnit.SECONDS);
+                     blockBeforeOpen.await(10, TimeUnit.SECONDS);
+                     return;
+                  } catch (Throwable ignored) {
+                  }
+               }
+               HashMap<Symbol, Object> brokerIDProperties = new HashMap<>();
+               brokerIDProperties.put(AMQPMirrorControllerSource.BROKER_ID, "fake-id");
+               x.setProperties(brokerIDProperties);
+               x.setOfferedCapabilities(new Symbol[]{AMQPMirrorControllerSource.MIRROR_CAPABILITY});
+               x.setTarget(x.getRemoteTarget());
+               x.open();
+               x.handler((del, msg) -> {
+                  if (msg.getApplicationProperties() != null) {
+                     Map map = msg.getApplicationProperties().getValue();
+                     Object value = map.get("sender");
+                     if (value != null) {
+                        if (messagesReceived.get() != ((Integer) value).intValue()) {
+                           logger.warn("Message out of order. Expected " + messagesReceived.get() + " but received " + value);
+                           errors.incrementAndGet();
+                        }
+                        messagesReceived.incrementAndGet();
+                     }
+                  }
+                  del.disposition(Accepted.getInstance(), true);
+                  if (x.getPrefetch() == 0) {
+                     x.flow(1);
+                  }
```

Comment:
```suggestion
```
The reciever auto-accept wasnt disabled, and prefetch wasnt set to 0, so you could just remove these and it will act the same.